### PR TITLE
GLTFLoader: Add createNodeMesh hook

### PR DESCRIPTION
**Pull request by takahirox**
**Background**

We added invokeAll `createNodeAttachment` hook in `GLTFLoader` https://github.com/mrdoob/three.js/pull/19892. It's a good hook for adding extra objects to a node object, like light.

But as I wrote in https://github.com/mrdoob/three.js/pull/19892#issue-454171225 this hook can't be used for suppressing core spec meshes (and cameras). Some extensions want to suppress loading core spec objects (and instead want to load objects defined in the extensions). eg. materials unlit extension

Actually I needed this capability in node as I was writing [`MSFT_lod` extension plugin](https://github.com/takahirox/three-gltf-extensions/tree/main/loaders/MSFT_lod). So I would like to add a new hook point for node. I know non `KHR_` extensions are not really our glTF loader extensibility mechanism targets, but it would be good if we can add a new hook point without messing the core code.

InvokeOne `loadNode()` hook may be one of the options similar to `loadMaterial` or others. But core `loadNode()` does [a lot common works for example make a group for multiple objects and setting transform](https://github.com/mrdoob/three.js/blob/r126/examples/js/loaders/GLTFLoader.js#L3637-L3705). I think extensions need the same ones.

**Suggestion**

So I would like to suggest to extract [the part creating node meshes](https://github.com/mrdoob/three.js/blob/r126/examples/js/loaders/GLTFLoader.js#L3582-L3609) and add an invokeOne hook for it, named `createNodeMesh` hook. The common works for node still be usable for the extensions using the new hook. And the extensions using the new hook doesn't have an effect to the extensions using `createNodeAttachment` like light extension. I think it's good for most of the cases.

### CODE REVIEW

**Comment by takahirox** [comment]
Sorry I know you seem busy but friendly ping @donmccurdy

**Review by donmccurdy — APPROVED** [review]
Looks good to me!

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -3581,28 +3614,9 @@ THREE.GLTFLoader = ( function () {
...
-								o.morphTargetInfluences[ i ] = nodeDef.weights[ i ];
-
-							}
-
-						} );
-
-					}
-
-					return node;
+					return ext.createNodeMesh && ext.createNodeMesh( nodeIndex );
```
Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js:3620]: Sorry, I found a problem in this implementation after the review has been done. Plugins' createNodeMesh() should be called even if core `nodeDef.mesh` is undefined. e.g. GPU instancing extension without fallback in core. I would update soon.

Comment (takahirox) [reply examples/js/loaders/GLTFLoader.js:3624]: Updated

**Comment by mrdoob** [comment]
Probably better to move it to next cycle so we have time to test.

**Comment by takahirox** [comment]
It seems this PR had conflicts in `examples/js/loaders/GLTFLoader.js` made by #21588. But `examples/jsm/loaders/GLTFLoader.js` had no conflicts so I ran `npm run build-examples` and now no conflicts but there are lot of diffs in `examples/js/loaders/GLTFLoader.js`. I haven't looked into the examples builder script but do the diffs look ok? I think if I trust the script, it should be fine.

**Comment by mrdoob** [comment]
We have no other option than trusting the script... 😅

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**